src/search.rs

```diff
@@ -0,0 +1,166 @@
+use crate::document::advanced::DocumentEncryptUnmanagedResult;
+use crate::document::advanced::*;
+use crate::document::DocumentEncryptOpts;
+use crate::internal::{take_lock, IronOxideErr};
+use crate::Result;
+use crate::{GroupId, IronOxide};
+use async_trait::async_trait;
+use rand::{self, RngCore};
+use std::convert::{TryFrom, TryInto};
+use std::ops::DerefMut;
+
+use ironcore_search_helpers::generate_hashes_for_string;
+
+///The required length of the salt.
+const REQUIRED_LEN: usize = 32;
+
+///The result of creating a new index as well as initializing a BlindIndexSearchSdk.
+///If you only want to create the index, see create_index.
+pub struct CreatedIndexResult {
+    pub encrypted_salt: EncryptedBlindIndexSalt,
+    pub sdk: BlindIndexSearchSdk,
+}
+
+pub struct EncryptedBlindIndexSalt {
+    pub encrypted_deks: Vec<u8>,
+    pub encrypted_salt_bytes: Vec<u8>,
+}
+
+///Trait which gives the ability to create an blind index.
+#[async_trait]
+pub trait BlindIndexSearchInitialize {
+    ///Given the encrypted blind index salt, decrypt it and give back the BlindIndexSearchSdk object.
+    async fn initialize_search(
+        &self,
+        search: &EncryptedBlindIndexSalt,
+    ) -> Result<BlindIndexSearchSdk>;
+    ///Create an index and encrypt it to the provided group_id.
+    ///If you need to index data immediately, see `create_index_and_initialize_search`.
+    async fn create_index(&self, group_id: &GroupId) -> Result<EncryptedBlindIndexSalt>;
+    ///Create an index, encrypt it and initialize a BlindIndexSearchSdk for immediate use.
+    async fn create_index_and_initialize_search(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<CreatedIndexResult>;
+}
+
+#[async_trait]
+impl BlindIndexSearchInitialize for IronOxide {
+    async fn initialize_search(
```

Comment:
consider rename to `initialize_blind_index_search`?

Reply:
ok.

Reply:
You said "ok", but didn't change it. Intentional?

Reply:
Sorry. Pushed.